mbtools/copy_resources_s3.py

```diff
@@ -1,173 +1,149 @@
-        data = json.load(f)
-        for item in data:
-            hashes.append(item["sha1"])
-    return hashes
+    with open(template_path) as template_file:
+        template = environment.from_string(template_file.read())
+        with open(f'{index_file}', mode='w') as workflow_file:
+            workflow_file.write(template.render(data=data))
 
 
-def new_resource_hashes(resource_dir):
+def resource_hashes(resource_dir):
     """Generates sha1 hashes for all the resources in a local directory"""
+
     sha1_map = {}
-    duplicates = []
     buff_size = io.DEFAULT_BUFFER_SIZE
+    path_to_files = []
+    for (dir_path, dir_names, file_names) in os.walk(resource_dir):
+        for file in file_names:
+            path_to_files.append(os.path.join(dir_path, file))
+
     resource_files = filter(
-        lambda x: x not in IGNORED_FILES,
-        os.listdir(resource_dir)
+        lambda x: Path(x).stem not in IGNORED_FILES,
+        path_to_files
     )
 
-    for filename in resource_files:
-
+    for full_path in resource_files:
+        mime_type = get_mime_type(full_path)
         sha1 = hashlib.sha1()
-        full_path = os.path.join(resource_dir, filename)
         with open(full_path, 'rb') as f:
             while True:
                 data = f.read(buff_size)
                 if not data:
                     break
                 sha1.update(data)
-        if sha1.hexdigest() not in sha1_map:
-            sha1_map[sha1.hexdigest()] = full_path
-        else:
-            duplicates.append([sha1.hexdigest(), full_path])
-    return sha1_map, duplicates
+            sha1_map[sha1.hexdigest()] = {
+                'path': full_path,
+                'mime_type': mime_type
+            }
+
+    return sha1_map
 
 
-def get_mime_type(filename):
+def get_mime_type(filepath):
     """Get the MIME type of file with libmagic """
     mime_type = ''
     try:
-        mime_type = magic.from_file(filename, mime=True)
+        mime_type = magic.from_file(filepath, mime=True)
     finally:
         return mime_type
 
 
-def add_new_resources_to_s3(bucket, s3_dir, hashes, hash2filename_map,
-                            metadata_file
-                            ):
-    """Add the files specified in filename_map to s3 iff they their
+def add_new_resources_to_s3(bucket, s3_dir, hashes_to_update,
+                            hash_to_filedata_map):
+    """Add the files specified in hashes_to_update to s3 if they their
     corresponding sha1 key exists in hashes"""
 
     s3_client = boto3.client("s3")
-    new_metadata = []
-    for hash_key in hashes:
+    metadata = []
+    for hash_key in hashes_to_update:
         full_keypath = s3_dir + '/' + hash_key
 
         try:
             s3_client.head_object(Bucket=bucket,
                                   Key=full_keypath)
-            filename = os.path.basename(hash2filename_map[hash_key])
+            filename = os.path.basename(hash_to_filedata_map[hash_key]['path'])
```

Comment:
I realize it wasn't covered earlier, but it would be nice to add test coverage for this code path as part of this PR.